File: stack.java
import java.util.*;

public class stack {
    stacknode top;

    class stacknode {
        int val;
        stacknode next;

        stacknode(int val) {
            this.val = val;
            this.next = null;
        }
    }

    boolean isempty() {
        if (top == null) {
            return true;
        }
        return false;
    }

    void push(int val) {
        stacknode ptr = new stacknode(val);
        ptr.next = top;
        top = ptr;
    }

    void pop() {
        if (isempty()) {
            System.out.println("underflow");
            return;
        } else {
            top = top.next;
        }
    }

    void display() {
        stacknode temp = top;
        while (temp != null) {
            System.out.print(temp.val + " ");
            temp = temp.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        stack ls = new stack();
        ls.pop();
        ls.push(3);
        ls.push(2);
        ls.push(1);
        ls.display();
        ls.pop();
        ls.display();
        Stack<Integer> obj = new Stack<>();
        obj.push(5);
        obj.push(50);
        for (Integer i : obj) {
            System.out.print(i + " ");
        }
        System.out.print(obj.pop());
    }
}

File: stack.c
#include<stdio.h>
#include<stdlib.h>

struct stack
{
    int val;
    struct stack *next;
};

struct stack *top=NULL;

int isempty(){
    if(top==NULL){
        return 1;
    }
    return 0;
}

void push(int val){
      struct stack *ptr=(struct stack *)malloc(sizeof(struct stack));
      ptr->val=val;
      if(isempty()){
        ptr->next=NULL;
        top=ptr;
        return;
      }else{
          ptr->next=top;
          top=ptr;
      }
}

void pop(){
    if(isempty()){
        printf("underflow");
        return;
    }
    top=top->next;
}

void display(){
    struct stack *temp=top;
    while(temp!=NULL){
        printf("%d ",temp->val);
        temp=temp->next;
    }
    printf("\n");
}
int main(){
      pop();
      display();
      push(10);
      display();
      push(20);
      push(30);
      display();
      push(40);
      display();
      pop();
      display();
}